perun-db/postgres.sql

```diff
@@ -1772,6 +1757,125 @@ alter table user_ext_source_attr_values add constraint uesattrval_attr_fk foreig
+
+CREATE TABLE host_attr_u_values (
+	host_id INT NOT NULL,
+	attr_id INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (host_id,attr_id) REFERENCES host_attr_values ON DELETE CASCADE
+);
+CREATE INDEX hauv_idx ON host_attr_u_values (host_id, attr_id);
+
+CREATE TABLE member_attr_u_values (
+	member_id INT NOT NULL,
+	attr_id INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (member_id,attr_id) REFERENCES member_attr_values ON DELETE CASCADE
+);
+CREATE INDEX mauv_idx ON member_attr_u_values (member_id, attr_id);
+
+CREATE TABLE member_group_attr_u_values (
+	member_id INT NOT NULL,
+	group_id INT NOT NULL,
+	attr_id INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (member_id,group_id,attr_id) REFERENCES member_group_attr_values ON DELETE CASCADE
+);
+CREATE INDEX mgauv_idx ON member_group_attr_u_values (member_id, group_id, attr_id);
+
+CREATE TABLE member_resource_attr_u_values (
+	member_id INT NOT NULL,
+	resource_id INT NOT NULL,
+	attr_id INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (member_id,resource_id,attr_id) REFERENCES member_resource_attr_values ON DELETE CASCADE
+);
+CREATE INDEX mrauv_idx ON member_resource_attr_u_values (member_id, resource_id, attr_id);
+
+CREATE TABLE resource_attr_u_values (
+	resource_id INT NOT NULL,
+	attr_id INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (resource_id,attr_id) REFERENCES resource_attr_values ON DELETE CASCADE
+);
+CREATE INDEX rauv_idx ON resource_attr_u_values (resource_id, attr_id);
+
+CREATE TABLE user_attr_u_values (
+	user_id  INT NOT NULL,
+	attr_id  INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (user_id,attr_id) REFERENCES user_attr_values ON DELETE CASCADE
+);
+CREATE INDEX uauv_idx ON user_attr_u_values (user_id, attr_id);
+
+CREATE TABLE user_ext_source_attr_u_values (
+	user_ext_source_id INT NOT NULL,
+	attr_id    INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (user_ext_source_id,attr_id) REFERENCES user_ext_source_attr_values ON DELETE CASCADE
+);
+CREATE INDEX uesauv_idx ON user_ext_source_attr_u_values (user_ext_source_id, attr_id);
+
+CREATE TABLE user_facility_attr_u_values (
+	user_id INT NOT NULL,
+	facility_id INT NOT NULL,
+	attr_id INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (user_id,facility_id,attr_id) REFERENCES user_facility_attr_values ON DELETE CASCADE
+);
+CREATE INDEX ufauv_idx ON user_facility_attr_u_values (user_id, facility_id, attr_id) ;
+
+CREATE TABLE vo_attr_u_values (
+	vo_id INT NOT NULL,
+	attr_id INT NOT NULL,
+	attr_value VARCHAR(4000),
+	UNIQUE (attr_id, attr_value),
+	FOREIGN KEY (vo_id,attr_id) REFERENCES vo_attr_values ON DELETE CASCADE
+);
+CREATE INDEX vauv_idx ON vo_attr_u_values (vo_id, attr_id) ;
```

Comment:
You should preserve existing order in the file or change it whole, not just part of it.